### PR TITLE
[BottomAppBar] Fix path animation

### PR DESCRIPTION
**Pull request by codeman7**
### Context
As of #5155 our animation didn't have an equal number of points in the bezierPath when there was or wasn't a FAB. When doing an animation between two `CGPath`s the points need to be the same in order to animate correctly.
### The problem
There were not an equal number of points between both paths so the animation didn't know how to map the values in each path to each other.
### The fix
Update the path to render correctly by adding a point to the `withoutCut` function.
### Related bug
#5603 
### Videos
| Before | After |
| - | - |
|before|after|

**_note_** Example Code snippet

```objc
// In MDCBottomAppBarAttributes.h
kMDCFloatingButtonExitDuration = 2.180
kMDCFloatingButtonEnterDuration = 2.270

// In example file
[bottomAppBar.floatingButton removeFromSuperview];
// toggle between
BOOL animate = true
[bottomAppBar setFloatingButtonHidden:animate animated:YES];
animate = !animate
```

### CODE REVIEW

**Review by jverkoey — APPROVED** [review]
What a fun before animation!

**Comment by jverkoey** [comment]
We may be able to write a unit test for this by using CGPathApply to verify that the paths contain an equal number of points in the animation: https://oleb.net/blog/2015/06/c-callbacks-in-swift/

File: components/BottomAppBar/src/private/MDCBottomAppBarLayer.m [hunk 1]
```diff
@@ -102,12 +101,12 @@ - (UIBezierPath *)drawWithPlainPath:(UIBezierPath *)bottomBarPath
...
-                   hypotenuseLength:(CGFloat)hypotenuseLength {
-  CGFloat halfOfHypotenuseLength = hypotenuseLength / 2;
+                          arcRadius:(CGFloat)arcRadius {
   [bottomBarPath moveToPoint:CGPointMake(0, yOffset)];
-  [bottomBarPath addLineToPoint:CGPointMake(arcCenter.x - halfOfHypotenuseLength, yOffset)];
+  [bottomBarPath addLineToPoint:CGPointMake(arcCenter.x - arcRadius, yOffset)];
   [bottomBarPath addLineToPoint:CGPointMake(arcCenter.x, yOffset)];
-  [bottomBarPath addLineToPoint:CGPointMake(arcCenter.x + halfOfHypotenuseLength, yOffset)];
+  [bottomBarPath addLineToPoint:CGPointMake(arcCenter.x + arcRadius, yOffset)];
+  [bottomBarPath addLineToPoint:CGPointMake(arcCenter.x + arcRadius, yOffset)];
```
Comment (romoore) [inline components/BottomAppBar/src/private/MDCBottomAppBarLayer.m:127]: If these are going to be the exact same lines, please add a comment to remark that they are intentional (and explain why) and not an accidental line paste.  Maintainers are likely to see this as a no-op and attempt to remove it.

**Review by romoore — APPROVED** [review]

**Comment by codeman7** [comment]
This was merged in order to get into this weeks release. The test still need some refactoring, a bug has been opened to do so, that bug can be found [here](https://github.com/material-components/material-components-ios/issues/5633).

**Pull request merged**